enginetest/queries/insert_queries.go

```diff
@@ -831,6 +831,25 @@ var SpatialInsertQueries = []WriteQueryTest{
 }
 
 var InsertScripts = []ScriptTest{
+	{
+		// https://github.com/dolthub/dolt/issues/7322
+		Name: "issue 7322: values expression is subquery",
+		SetUpScript: []string{
+			"create table xy (x int auto_increment primary key, y varchar(50) not null)",
+			"create table uv (u int auto_increment primary key, v varchar(50) not null, x_id int, constraint u_x_fk foreign key (x_id) references xy (x))",
+			"insert into xy values (1,'admin'), (2, 'standard')",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "INSERT INTO uv(v, x_id) VALUES ('test', (SELECT x FROM xy WHERE y = 'admin'));",
```

Comment:
Would be nice to have a test of an error case here too -- what if there is more than one result from the subquery?